doc/book/box/indexes.rst

```diff
@@ -24,46 +20,454 @@ The first index defined on a space is called the **primary key index**,
+            - ALL, EQ, GT
+            - ALL, EQ, GT, GE, LT, LE, OVERLAPS, NEIGHBOR
+            - ALL, EQ, BITS_ALL_SET, BITS_ANY_SET, BITS_ALL_NOT_SET
+
+********************************************************************************
+TREE indexes
+********************************************************************************
+
+The default index type is 'TREE'.
+TREE indexes are provided by memtx and vinyl engines, can index unique and
+non-unique values, support partial key searches, comparisons and ordered results.
+
+This is a universal type of indexes, for most cases it will be the best choice.
+
+Additionally, memtx engine supports HASH, RTREE and BITSET indexes.
+
+********************************************************************************
+HASH indexes
+********************************************************************************
+
+HASH indexes require unique fields and loses to TREE in almost all respects.
+So we do not recommend to use it in the applications.
+HASH is now present in Tarantool mainly because of backward compatibility.
+
+Here are some tips. Do not use HASH index:
+
+* just if you want to
+* if you think that HASH is faster with no performance metering
+* if you want to iterate over the data
+* for primary key
+* as an only index
+
+Use HASH index:
+
+* if it is a secondary key
+* if you 100% won't need to make it non-unique
+* if you really need that 2-5% performance improvement
+* if you have taken measurements on your data and you see an increase in performance
+* if you save every byte on tuples (HASH is a little more compact)
+
+********************************************************************************
+RTREE indexes
+********************************************************************************
+
+RTREE is a multidimensional index supporting up to 32 dimensions.
+It is used especially for indexing spatial information, such as geographical
+objects. In :ref:`this example <box_index-rtree>`
+we demonstrate spatial searches via RTREE index.
+
+RTREE index can accept two types of ``distance`` functions:
+``euclid`` and ``manhattan``.
+
+**Example:**
+
+..  code-block:: tarantoolsession
+
+    tarantool> box.schema.space.create('rtree_example')
+    tarantool> box.space.rtree_example:create_index('primary')
+    tarantool> box.space.rtree_example:create_index('rtree',{unique=false,type='RTREE', parts={2,'ARRAY'}})
+    tarantool> box.space.rtree_example:insert{1, {3, 5, 9, 10}}
+    tarantool> box.space.rtree_example:insert{2, {10, 11}}
+    tarantool> box.space.rtree_example.index.rtree:select({4, 7, 5, 9}, {iterator = 'GT'})
+
+The result will be:
+
+..  code-block:: tarantoolsession
+
+    ---
+    - - [1, [3, 5, 9, 10]]
+    ...
+
+because a rectangle whose corners are at coordinates ``4,7,5,9`` is entirely
+within a rectangle whose corners are at coordinates ``3,5,9,10``.
+
+********************************************************************************
+BITSET indexes
```

Comment:
At this page https://tarantool-develop.herokuapp.com/en/doc/latest/book/box/data_model/ we have

![image](https://user-images.githubusercontent.com/8830475/108992738-b221a780-76aa-11eb-8a74-4491c2a7f77c.png)

Actually it's not so after https://github.com/tarantool/tarantool/issues/5071